src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2352,51 +2434,105 @@ private void ReportNullabilityMismatchWithTargetDelegate(SyntaxNode syntax, Name
+                        if (!conversion.IsValid)
                         {
-                            // PROTOTYPE(NullableReferenceTypes): Conversions: Lifted operator
+                            canConvert = false;
                             break;
                         }
-                        // PROTOTYPE(NullableReferenceTypes): Update method based on operandType.
-                        var resultType = methodOpt.ReturnType;
-                        // Ensure converted type matches expected target type, specifically for lifted
-                        // conversions. See LocalRewriter.MakeConversionNode for similar handling
-                        // based on conversion.BestUserDefinedConversionAnalysis.ToType.
-                        if (!fromConversionNode)
-                        {
-                            if (resultType.TypeSymbol != targetType)
-                            {
-                                // PROTOTYPE(NullableReferenceTypes): Should handle nested nullability
-                                // and top-level nullability. Conversion from resultType to targetType
-                                // should be treated as a built-in conversion.
-                                resultType = TypeSymbolWithAnnotations.Create(targetType);
-                            }
-                        }
-                        return resultType;
+
+                        // operand -> conversion "from" type
+                        operandType = ApplyConversion(
+                            node,
+                            operandOpt,
+                            conversion.UserDefinedFromConversion,
+                            conversion.BestUserDefinedConversionAnalysis.FromType,
+                            operandType,
+                            checkConversion: false,
+                            fromExplicitCast: false,
+                            out _);
+
+                        // PROTOTYPE(NullableReferenceTypes): Update method based on operandType
+                        // (see StaticNullChecking_FlowAnalysis.Conversions_07).
+                        var methodOpt = conversion.Method;
+                        Debug.Assert((object)methodOpt != null);
+                        Debug.Assert(methodOpt.ParameterCount == 1);
+                        var parameter = methodOpt.Parameters[0];
+
+                        // conversion "from" type -> method parameter type
+                        operandType = ClassifyAndApplyConversion(node, parameter.Type.TypeSymbol, operandType);
+
+                        // method parameter type -> method return type
+                        ReportArgumentWarnings(operandOpt, operandType, parameter);
+                        operandType = methodOpt.ReturnType;
+
+                        // method return type -> conversion "to" type
+                        operandType = ClassifyAndApplyConversion(node, conversion.BestUserDefinedConversionAnalysis.ToType, operandType);
```

Comment:
>ClassifyAndApplyConversion [](start = 38, length = 26)

What kind of changes and diagnostics this call is responsible for?

Reply:
It is responsible for converting nested nullability and reporting warnings on nested nullability mismatch.

---
In reply to: [190746518](https://github.com/dotnet/roslyn/pull/26356#discussion_r190746518) [](ancestors = 190746518)